### PR TITLE
Feature/text aug mapper

### PR DESCRIPTION
**Pull request by HYLcool**
- **New OPs**: simple text augmentation OPs for en/zh; aggregate several augmentation methods in each OP
  -  They are also updated in the `config_all.yaml` with augmentation examples.
- **Config Optimization**:
  - A new argument `export_in_parallel` to control whether to export the result dataset into a single file in parallel. It's disabled by default.
  - The input config file will be backed up in the work directory for further usage.
  - Data-Juicer will display the config table before starting the processing for better management of the current running tasks.

### CODE REVIEW

**Review by zhijianma — APPROVED** [review]

File: data_juicer/core/tracer.py [hunk 1]
```diff
@@ -76,6 +76,45 @@ def trace_mapper(self, op_name: str, previous_ds: Dataset,
...
+        Compare datasets before and after a BatchMapper.
+
+        This will mainly show the new samples augmented by the BatchMapper
+
+        :param op_name: the op name of mapper
+        :param previous_ds: dataset before the mapper process
+        :param processed_ds: dataset processed by the mapper
+        :return:
+        """
+        assert previous_ds[0]['text'] == processed_ds[0]['text']
```
[the remark this answers is not in the document]
Comment (yxdyc) [reply data_juicer/core/tracer.py:91]: +1

File: data_juicer/ops/base_op.py [hunk 2]
```diff
@@ -20,13 +20,37 @@ def __init__(self, text_key: str = None):
...
         """
-        For sample level, sample --> sample or sample --> [samples, ...]
+        For sample level, sample --> sample
 
         :param sample: sample to process
         :return: processed sample
         """
         raise NotImplementedError
 
+class BatchMapper(Mapper):
```
Comment (yxdyc) [inline data_juicer/ops/base_op.py:30]: There is no substantial difference between `Mapper` and `BatchMapper` classes in terms of signature and implementation. In other words, there is no need for us to introduce `BatchMapper`?

**Review by yxdyc — COMMENTED** [review]
Good job. Plz see the inline comments

File: configs/config_all.yaml [hunk 3]
```diff
@@ -5,11 +5,12 @@
...
 dataset_path: '/path/to/your/dataset'                       # path to your dataset directory or file with weights(0.0-1.0), 1.0 as default.
-                                                            # Accepted format: 'weight1(optional) dataset1-path weight2(optional) dataset2-path '
+                                                            # Accepted format: 'weight1(optional) dataset1-path weight2(optional) dataset2-path
 export_path: '/path/to/result/dataset.jsonl'                # path to processed result dataset. Supported suffixes include ['jsonl', 'json', 'parquet']
 export_shard_size: 0                                        # Shard size of exported dataset in Byte. In default, it's 0, which means export the whole dataset into only one file. If it's set a positive number, the exported dataset will be split into several dataset shards, and the max size of each shard won't larger than the export_shard_size
+export_in_parallel: false                                   # Whether to export the result dataset in parallel to a single file. It only works when export_shard_size is 0, and its default number of processes is the same as the argument np.
```
[the remark this answers is not in the document]
Comment (HYLcool) [reply configs/config_all.yaml:11]: "False" can ensure the stability of exporting. Sometimes, "True" might require much more time due to the IO blocking (e.g. exporting a very large dataset to the NAS with very high concurrency). But "True" speeds up the exporting indeed for most situations.

File: data_juicer/ops/base_op.py (hunk 2 above)
Comment (HYLcool) [reply data_juicer/ops/base_op.py:30]: That's right.

**Review by chenhesen — APPROVED** [review]

**Review by yxdyc — APPROVED** [review]

**Pull request merged**